ovs-p4rt/sidecar/newovsp4rt.cc

```diff
@@ -2110,6 +2110,49 @@ absl::StatusOr<::p4::v1::ReadResponse> GetTxAccVsiTableEntry(
   return client.sendReadRequest(read_request);
 }
 
+absl::StatusOr<uint32> GetTxAccVsiPort(ClientInterface& client,
+                                       const ::p4::config::v1::P4Info& p4info,
+                                       uint32 src_port) {
+  auto response_or_status = GetTxAccVsiTableEntry(client, src_port, p4info);
+  auto status = response_or_status.status();
+  if (!status.ok()) return status;
+
+  ::p4::v1::ReadResponse read_response = std::move(response_or_status).value();
+
+  // TODO(derek): remove dead code
+  std::vector<::p4::v1::TableEntry> table_entries;
+  table_entries.reserve(read_response.entities().size());
+
+  int param_id =
+      GetParamId(p4info, TX_ACC_VSI_TABLE_ACTION_L2_FWD_AND_BYPASS_BRIDGE,
+                 ACTION_L2_FWD_AND_BYPASS_BRIDGE_PARAM_PORT);
+
+  uint32_t host_sp = 0;
+  for (const auto& entity : read_response.entities()) {
+    ::p4::v1::TableEntry table_entry = entity.table_entry();
+    // TODO(derek): why mutable?
+    auto* table_action = table_entry.mutable_action();
+    // TODO(derek): why mutable?
+    auto* action = table_action->mutable_action();
+    // TODO(derek): validate table_action->preamble.id?
+    for (const auto& param : action->params()) {
+      if (param_id == param.param_id()) {
+        const std::string& s1 = param.value();
+        // TODO(derek): why make a mutable copy?
+        std::string s2 = s1;
+        for (int param_bytes = 0; param_bytes < 4; param_bytes++) {
+          host_sp = host_sp << 8 | int(s2[param_bytes]);
+        }
+        // TODO(derek): This break exits to the outermost for loop,
+        // advancing to the next table entry. Is this correct?
+        // Would `return host_sp` be better?
```

Comment:
If we replace `break` with `return host_sp`, we can (1) define and initialize `host_sp` within the `if` statement, and (2) change line 2153 to `return 0`.